docs/transitive-dependencies-management.md

```diff
@@ -0,0 +1,68 @@
+# Maven
+
+Maven downloads dependency of [the latest version](https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges)
+matching the declared range by default, in other words whenever new versions of Selenium 4 libraries are published
+they are pulled transitively as Appium Java Client dependencies at the first project (re)build automatically.
+
+In order to pin Selenium dependencies they should be declared in `pom.xml` in the following way:
+
+```xml
+<dependencies>
+    <dependency>
+        <groupId>io.appium</groupId>
+        <artifactId>java-client</artifactId>
+        <version>X.Y.Z</version>
+        <exclusions>
+            <exclusion>
+                <groupId>org.seleniumhq.selenium</groupId>
+                <artifactId>selenium-api</artifactId>
+            </exclusion>
+            <exclusion>
+                <groupId>org.seleniumhq.selenium</groupId>
+                <artifactId>selenium-remote-driver</artifactId>
+            </exclusion>
+            <exclusion>
+                <groupId>org.seleniumhq.selenium</groupId>
+                <artifactId>selenium-support</artifactId>
+            </exclusion>
+        </exclusions>
+    </dependency>
+    <dependency>
+        <groupId>org.seleniumhq.selenium</groupId>
+        <artifactId>selenium-api</artifactId>
+        <version>A.B.C</version>
+    </dependency>
+    <dependency>
+        <groupId>org.seleniumhq.selenium</groupId>
+        <artifactId>selenium-remote-driver</artifactId>
+        <version>A.B.C</version>
+    </dependency>
+    <dependency>
+        <groupId>org.seleniumhq.selenium</groupId>
+        <artifactId>selenium-support</artifactId>
+        <version>A.B.C</version>
+    </dependency>
+</dependencies>
+```
+
+# Gradle
+
+Gradle uses [Module Metadata](https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html)
+to perform improved dependency resolution whenever it is available. Gradle Module Metadata for Appium Java Client is
+published automatically with every release and is available on Maven Central.
+
+Appium Java Client declares [preferred](https://docs.gradle.org/current/userguide/rich_versions.html#rich-version-constraints)
+Selenium dependencies version which is equal to the lowest boundary in the version range, i.e. the lowest compatible
+Selenium dependencies are pulled by Gradle by default. It's strictly recommended to do not use versions lower than the
+range boundary, because unresolvable compilation and runtime errors may occur.
+
+In order to use newer Selenium dependencies they should be explicitly added to Gradle build script (`build.gradle`):
+
+```gradle
+dependencies {
+    implementation('io.appium:java-client:X.Y.Z')
```

Comment:
shouldn't we also exclude selenium here like we did for maven?

Reply:
nope, it's not needed.
since we can add Selenium of the same version as the lowest boundary or of the higher version, Gradle will resolve this case automatically and dependencies with highest version will be chosen.
an attempt to use Selenium version lower than declared in the range will lead to build error

Reply:
```gradle
dependencies {
    implementation('io.appium:java-client:8.3.0')
    implementation('org.seleniumhq.selenium:selenium-support:4.5.0')
}
```
results in
```
...
org.seleniumhq.selenium:selenium-api:4.7.0
...
   Selection reasons:
      - By conflict resolution : between versions 4.5.0 and 4.7.0

org.seleniumhq.selenium:selenium-api:4.7.0
+--- org.seleniumhq.selenium:selenium-http:4.7.0
...
org.seleniumhq.selenium:selenium-api:4.5.0 -> 4.7.0
...

org.seleniumhq.selenium:selenium-api:{strictly [4.7.0, 5.0); prefer 4.7.0} -> 4.7.0
\--- io.appium:java-client:8.3.0
     \--- compileClasspath

```

Reply:
Here is an example with an attempt to force it:
```gradle
dependencies {
    implementation('io.appium:java-client:8.3.0')
    implementation('org.seleniumhq.selenium:selenium-support:4.5.0') {
        force = true
    }
}
```

```
Execution failed for task ':lib:test'.
> Could not resolve all files for configuration ':lib:testRuntimeClasspath'.
   > Could not resolve org.seleniumhq.selenium:selenium-support:4.5.0.
     Required by:
         project :lib
      > Cannot find a version of 'org.seleniumhq.selenium:selenium-support' that satisfies the version constraints:
           Dependency path 'gradle-test:lib:unspecified' --> 'org.seleniumhq.selenium:selenium-support:4.5.0'
           Dependency path 'gradle-test:lib:unspecified' --> 'io.appium:java-client:8.3.0' (runtimeElements) --> 'org.seleniumhq.selenium:selenium-support:{strictly [4.7.0, 5.0); prefer 4.7.0}'

   > Could not resolve org.seleniumhq.selenium:selenium-support:{strictly [4.7.0, 5.0); prefer 4.7.0}.
     Required by:
         project :lib > io.appium:java-client:8.3.0
      > Cannot find a version of 'org.seleniumhq.selenium:selenium-support' that satisfies the version constraints:
           Dependency path 'gradle-test:lib:unspecified' --> 'org.seleniumhq.selenium:selenium-support:4.5.0'
           Dependency path 'gradle-test:lib:unspecified' --> 'io.appium:java-client:8.3.0' (runtimeElements) --> 'org.seleniumhq.selenium:selenium-support:{strictly [4.7.0, 5.0); prefer 4.7.0}'
```

Reply:
and finally
```gradle
dependencies {
    implementation('io.appium:java-client:8.3.0')
    implementation('org.seleniumhq.selenium:selenium-support:4.7.2')
}
```
results in
```
org.seleniumhq.selenium:selenium-api:4.7.2
...
   Selection reasons:
      - By conflict resolution : between versions 4.7.2 and 4.7.0

org.seleniumhq.selenium:selenium-api:4.7.2
...

org.seleniumhq.selenium:selenium-api:{strictly [4.7.0, 5.0); prefer 4.7.0} -> 4.7.2
\--- io.appium:java-client:8.3.0
     \--- compileClasspath

```